### PR TITLE
Updated unit tests in src/protocols/bdx/tests/ to use PW instead of NL.

### PR DESCRIPTION
**Pull request by feasel0**
These changes are replacing nl-unit tests with the pigweed-unit tests, continuing the efforts from [#29342](https://github.com/project-chip/connectedhomeip/pull/29342) and others.  I updated the tests in src/protocols/bdx/tests/  No changes were made to the logic of the tests.

### CODE REVIEW

File: src/protocols/bdx/tests/TestBdxTransferSession.cpp [hunk 1]
```diff
@@ -189,14 +184,13 @@ void SendAndVerifyTransferInit(nlTestSuite * inSuite, void * inContext, Transfer
...
-                    inSuite,
-                    !memcmp(initData.Metadata, outEvent.transferInitData.Metadata, outEvent.transferInitData.MetadataLength));
+                EXPECT_EQ(0,
+                          memcmp(initData.Metadata, outEvent.transferInitData.Metadata, outEvent.transferInitData.MetadataLength));
             }
         }
         else
         {
-            NL_TEST_ASSERT(inSuite, false); // Metadata length mismatch
+            EXPECT_TRUE(false); // Metadata length mismatch
```
Comment (arkq) [inline src/protocols/bdx/tests/TestBdxTransferSession.cpp:193]: Refactor test logic to get rid of `ASSERT_TRUE(false)`. Later in the test output such information ("false != true") will be meaningless.

**Review by arkq — CHANGES_REQUESTED** [review]
Refactor tests and correctly use `ASSERT_*` in paces where test should be stopped immediately.

File: src/protocols/bdx/tests/TestBdxTransferSession.cpp (hunk 1 above)
Comment (feasel0) [reply src/protocols/bdx/tests/TestBdxTransferSession.cpp]: Changed in all places that seemed applicable.

File: src/protocols/bdx/tests/TestBdxTransferSession.cpp [hunk 2]
```diff
@@ -91,55 +89,53 @@ CHIP_ERROR AttachHeaderAndSend(TransferSession::MessageTypeData typeData, chip::
...
 // The msg argument is expected to begin at the message data start, not at the PayloadHeader.
-void VerifyStatusReport(nlTestSuite * inSuite, void * inContext, const System::PacketBufferHandle & msg, StatusCode expectedCode)
+void VerifyStatusReport(const System::PacketBufferHandle & msg, StatusCode expectedCode)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     if (msg.IsNull())
     {
-        NL_TEST_ASSERT(inSuite, false);
+        EXPECT_TRUE(false);
```
[the remark this answers is not in the document]
Comment (feasel0) [reply src/protocols/bdx/tests/TestBdxTransferSession.cpp]: Changed in all places that seemed applicable.

File: src/protocols/bdx/tests/TestBdxMessages.cpp [hunk 3]
```diff
@@ -17,29 +16,40 @@ using namespace chip::bdx;
...
 
-void TestTransferInitMessage(nlTestSuite * inSuite, void * inContext)
+struct TestBdxMessages : public ::testing::Test
+{
+    static void SetUpTestSuite()
+    {
+        CHIP_ERROR error = chip::Platform::MemoryInit();
+        EXPECT_EQ(error, CHIP_NO_ERROR);
+    }
+
```
[the remark this answers is not in the document]
Comment (feasel0) [reply src/protocols/bdx/tests/TestBdxMessages.cpp]: Done.  Would it make sense to use ASSERT_EQ in the SetUp function so the test will not proceed if the memory initialization fails?

**Review by arkq — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**